orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -74,7 +74,8 @@ void AddRepeatedOutputs(
 static void checkSendInputTensorElemTypes(
     InferenceContext& ctx,
     const std::string& attributeName,
-    const size_t inputSize) {
+    const size_t inputSize,
+    const size_t starting_idx_offset = 2) {
```

Comment:
This looks quite brittle -- is there any checks we can add in debug builds to confirm the correct value is used?